test/test_parser.rb

```diff
@@ -5336,4 +5337,21 @@ def test_bug_regex_verification
       s(:regexp, s(:str, "#)"), s(:regopt, :x)),
       %Q{/#)/x})
   end
+
+  def test_bug_non_utf8_symbol_literal
+    # "\xC7" is actually invalid in US-ASCII, but Ruby allows the conversion of
+    # this invalid string to a symbol, regardless if you do so using
+    # `String#to_sym` or if you have :"\xC7" appear as a literal in a US-ASCII
+    # encoded source file.
+    #
+    # If it was UTF-8, we would get an EncodingError.
+
+    bad_symbol = "\xC7".force_encoding(Encoding::US_ASCII).to_sym
```

Comment:
This is actually incorrect, because:
```
# -*- encoding: us-ascii -*-

p :"\xC3".to_s # "\xC3"
p :"\xC3".to_s.valid_encoding? # true
p :"\xC3".to_s.encoding # #<Encoding:ASCII-8BIT>
```